code/datums/components/bump_attack.dm

```diff
@@ -69,33 +69,47 @@
 
 /datum/component/bump_attack/proc/human_bump_action(datum/source, atom/target)
 	SIGNAL_HANDLER
-	var/mob/living/carbon/human/bumper = parent
+	INVOKE_ASYNC(src, .proc/human_bump_action_checks, source, target)
```

Comment:
I'm not seing any sleep in the next procs, why are you using a wrapper function with an INVOKE ASYNC?

Reply:
There are sleeps in procs that are called further down.

Reply:
Then it should only invoke the proc that actually sleeps which I assume is human_do_bump_action